### DISCUSSION TITLE
Is it possible to partially fetch an embeddable?

### DISCUSSION THREAD

**Discussion by JeongJuhyeon** [mikro-orm/mikro-orm#3672] [Q&A] [answered]
Using 
- **mikro-orm 5.4.2**
- PostgreSQL
- ts-morph

**tl;dr** Trying to partially fetch embeddable but the properties aren't being selected, they don't appear in the generated query.

Entity and Embeddable:
```typescript
@Embeddable()
export class ReservationRestrictions {
  @Property()
  paxMin!: number;

  @Property()
  paxMax!: number;

  @Property()
  slotGapMinutes!: number;

  @Property()
  daysReservableInFuture!: number;

  @Property({ type: 'json', nullable: false })
  openingHours: Partial<Record<DayNumber, OpeningHours>> = {};

  constructor(dto: CreateStoreDto) {
    this.daysReservableInFuture =
      dto.reservationRestrictions.daysReservableInFuture;
    this.paxMax = dto.reservationRestrictions.paxMax;
    this.paxMin = dto.reservationRestrictions.paxMin;
    this.slotGapMinutes = dto.reservationRestrictions.slotGapMinutes;
    this.openingHours = dto.reservationRestrictions.openingHours;
  }
}

@Entity()
export class Store extends CustomBaseEntity {
  @OneToMany(() => Reservation, (reservation) => reservation.store)
  reservations = new Collection<Reservation>(this);

  @Embedded()
  storeInfo!: StoreInfo;

  @OneToOne(() => StoreOwner, (storeOwner) => storeOwner.store)
  owner!: StoreOwner;

  constructor(owner: StoreOwner, dto: CreateStoreDto) {
    super();
    this.owner = owner;
    this.reservationRestrictions = new ReservationRestrictions(dto);
    this.storeInfo = new StoreInfo(dto);
  }
}
```
The below call
```typescript
    this.storeRepository.findAll({
      fields: [
        'id',
        {
          storeInfo: ['name', 'subtitle', 'cuisine', 'pictures'],
        },
      ],
    });
```
Produces the following query:
`select "s0"."id" from "store" as "s0"`
I'm not sure if this is intentional or a bug. For reference, if I select the embeddable as a field (so all properties are pulled) it works fine:
```typescript
this.storeRepository.findAll({
      fields: ['id', 'storeInfo'],
    });
```
`select "s0"."id", "s0"."store_info_name", "s0"."store_info_subtitle", "s0"."store_info_pictures", "s0"."store_info_description", "s0"."store_info_address", "s0"."store_info_phone", "s0"."store_info_opening_hours", "s0"."store_info_city", "s0"."store_info_cuisine" from "store" as "s0"`

**Comment by B4nan** [comment, marked as answer]
Sounds like a bug, its generally not supported to use partial loading for object embeddables, for inline mode it should be probably possible, but we should definitely not ignore the input completely.

**Reply by JeongJuhyeon** [reply]
Sounds good, would be nice to change typings to prevent it (if possible) or at least throw a runtime error in the meantime and then maybe someday support it as a feature :)

**Reply by B4nan** [reply]
It is fixed in 5.5.1
